Add a Lexer for Bicep language

xaviermignot opened the pull request:
Add support for Bicep (Azure DSL language).

This is my first Lexer, and as I'm not a Ruby developer I am very open to feedback, I'll be happy to make the requested changes.
I have tried to follow the guidelines from the docs, but feel free to reach out if there is any broken rule (or any question).

Resolves #1887 (I hope 🤞)

flanakin commented:
@tancnle Any chance you could review this and consider it for the next release? This is a gap for my site right now 😞

xaviermignot commented:
Hello @tancnle, sorry for tagging you like this but this PR has been opened a while ago and is still waiting for review...
I'm in the same boat as @flanakin, I would very much bring syntax highlighting to the blog posts I have written about Bicep 🤗

Many thanks in advance, and as I said I'm open to feedback so don't hesitate if some change is required from my side !

xaviermignot commented:
Found and fixed the whitespace error returned by the linter and added some functions added by the Bicep team in the meantime.
New CI workflow is waiting to start ^^
